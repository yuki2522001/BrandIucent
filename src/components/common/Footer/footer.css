.footer {
  display: flex;
  justify-content: space-between;
  padding: 133px 94px 57px 313px;
  background: var(--border-primary);
  margin-top: 133px;
}

.footer__title {
  font-weight: 400;
  font-size: 17px;
  line-height: 22px;
  letter-spacing: -0.231818px;
  color: var(--text-secondary);
}

.footer__contact {
  display: flex;
}

.footer__item {
  margin-right: 240px;
}

.footer__about {
  display: flex;
  margin-top: 159px;
}

.footer__service {
  font-size: 15px;
  line-height: 20px;
  letter-spacing: -0.204545px;
  color: var(--light);
  margin-right: 100px;
}

.footer__social {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
}