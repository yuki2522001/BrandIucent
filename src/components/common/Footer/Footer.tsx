import React, { memo } from "react";
import { MENU_ABOUT, MENU_HELP, MENU_SERVICE } from "@constants/menu";
import MenuContact from "../MenuContact/MenuContact";
import url from "@assets/images/logos/logo.png";
import Logo from "../Logo/Logo";
import IconLink from "../IconLink/IconLink";
import ICON_LINK from "@constants/iconLink";
import "./footer.css";

const Footer: React.FC = memo(() => {
  return (
    <footer data-testid="footer" className="footer">
      <div className="footer__info">
        <div className="footer__contact">
          <div className="footer__item">
            <p className="footer__title">Services</p>
            <MenuContact menuList={MENU_SERVICE} />
          </div>
          <div className="footer__item">
            <p className="footer__title">About</p>
            <MenuContact menuList={MENU_ABOUT} />
          </div>
          <div className="footer__item">
            <p className="footer__title">Help</p>
            <MenuContact menuList={MENU_HELP} />
          </div>
        </div>
        <div className="footer__about">
          <p className="footer__service">Terms & Conditions</p>
          <p className="footer__service">Privacy Policy</p>
        </div>
      </div>
      <div className="footer__social">
        <Logo type="large" src={url} />
        <IconLink iconList={ICON_LINK} />
      </div>
    </footer>
  );
});

export default Footer;
