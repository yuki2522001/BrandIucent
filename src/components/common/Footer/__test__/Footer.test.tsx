import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import "@testing-library/jest-dom";
import Footer from "../Footer";

describe("Footer component", () => {
  test("should render Footer component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Footer />
      </Router>
    );
    expect(getByTestId("footer")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <Footer />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
