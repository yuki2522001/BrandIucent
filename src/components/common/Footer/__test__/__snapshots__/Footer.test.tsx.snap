// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
    data-testid="footer"
  >
    <div
      class="footer__info"
    >
      <div
        class="footer__contact"
      >
        <div
          class="footer__item"
        >
          <p
            class="footer__title"
          >
            Services
          </p>
          <div
            class="contact"
            data-testid="contact"
          >
            <ul
              class="contact__list"
            >
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Email Marketing
                </a>
              </li>
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Campaigns
                </a>
              </li>
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Branding
                </a>
              </li>
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Offline
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="footer__item"
        >
          <p
            class="footer__title"
          >
            About
          </p>
          <div
            class="contact"
            data-testid="contact"
          >
            <ul
              class="contact__list"
            >
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Our Story
                </a>
              </li>
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Benefits
                </a>
              </li>
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Team
                </a>
              </li>
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Careers
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="footer__item"
        >
          <p
            class="footer__title"
          >
            Help
          </p>
          <div
            class="contact"
            data-testid="contact"
          >
            <ul
              class="contact__list"
            >
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  FAQs
                </a>
              </li>
              <li
                class="contact__item"
              >
                <a
                  href="/"
                >
                  Contact Us
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="footer__about"
      >
        <p
          class="footer__service"
        >
          Terms & Conditions
        </p>
        <p
          class="footer__service"
        >
          Privacy Policy
        </p>
      </div>
    </div>
    <div
      class="footer__social"
    >
      <a
        data-testid="logo"
        href="/"
      >
        <img
          alt="This is logo"
          class="logo--large"
        />
      </a>
      <div
        class="icon__list"
        data-testid="icon-link"
      >
        <a
          class="icon__item"
          href="https://vi-vn.facebook.com/"
          title="Facebook"
        >
          <i
            class="icon-fb"
            data-testid="icon"
          />
        </a>
        <a
          class="icon__item"
          href="https://twitter.com/"
          title="Twitter"
        >
          <i
            class="icon-twitter"
            data-testid="icon"
          />
        </a>
        <a
          class="icon__item"
          href="https://www.instagram.com/"
          title="Instagram"
        >
          <i
            class="icon-instagram"
            data-testid="icon"
          />
        </a>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
